scripts/globals/items/random_ring.lua

```diff
@@ -0,0 +1,38 @@
+-----------------------------------
+-- ID: 15770
+-- Item: Random Ring
+-- Item Effect: DEX + 1~8
+-- Duration: 30 Minutes
+-----------------------------------
+local itemObject = {}
+
+itemObject.onItemCheck = function(target)
+    if target:getStatusEffect(xi.effect.ENCHANTMENT, nil, xi.items.RANDOM_RING) ~= nil then
+        target:delStatusEffect(xi.effect.ENCHANTMENT, nil, xi.items.RANDOM_RING)
+    end
+
+    return 0
+end
+
+itemObject.onItemUse = function(target)
+    if target:hasEquipped(xi.items.RANDOM_RING) then
+
+        if target:addStatusEffect(xi.effect.ENCHANTMENT, 0, 0, 1800, 0, 0, 0, xi.items.RANDOM_RING) then
+            local effect = target:getStatusEffect(xi.effect.ENCHANTMENT)
+            if effect then
+                local power = math.random(1, 8)
+
+                target:addMod(xi.mod.DEX, power)
```

Comment:
Check out memento_muffler.lua to do this in a different way. By adding a mod, it'll never go away

Reply:
the problem is that its a random roll for the dex 1-8 and when you zone or take the ring off the enchanment wears off. Can discuss

Reply:
that's exactly how memento muffler behaves. Just go
![image](https://github.com/AirSkyBoat/AirSkyBoat/assets/62350957/db009702-2805-460b-ba1a-6b1ea077cd16)